Fix NIC hotplug with `vhost_net=True`

saschalucas opened the pull request:
Since Qemu commit https://github.com/qemu/qemu/commit/db2a380c84574d8c76d7193b8af8535234fe5156 the QMP command `netdev_add` is enforced to have correct types. Effected versions are Qemu >=4.3 (Debian Bullseye and beyond). Fixing this by replacing the string with a Boolean.

This fix seems compatible down to Ubuntu Bionic (qemu-2.11) and should therefore be safe for backporting to 3.0.

Fixes #1651

rbott commented:
Thanks for uncovering this @saschalucas. LGTM!

rbott commented:
Just a quick sidenote: `vhost_net` seems to be completely undocumented in Ganeti. Maybe this PR could also add some details to the `gnt-instance` manpage?

We should even consider setting it to `true` as a sane default, but that should probably be at least a separate commit so we can choose to not bring that specific change into 3.0.2.

saschalucas commented:
> Just a quick sidenote: `vhost_net` seems to be completely undocumented in Ganeti. Maybe this PR could also add some details to the `gnt-instance` manpage?

Thanks for looking at that detail. I added a manual page entry.

> We should even consider setting it to `true` as a sane default, but that should probably be at least a separate commit so we can choose to not bring that specific change into 3.0.2.

At least for me, defaulting `vhost_net` to `True` makes sense. Qemu seems to ignore the resulting CLI option `vhost=on` for NIC types other than paravirtual (i.e. e1000). So I added a separate commit here. It may be noticed, that the new default only applies to new clusters (`gnt-cluster init`), because `config.data` values are filled with defaults at cluster creation time.

To make `py-tests` happy, it was necessary to add a new `test/data/kvm_5.2.0_help.txt`. I also added a symlink `test/data/kvm_current_help.txt` and referenced the symlink inside the unit test. This way, we can later switch all tests at once to new test data.

@rbott PTAL (re-approve).

saschalucas commented on man/gnt-instance.rst:
```diff
@@ -912,6 +912,17 @@ kvm\_path
+    through userspace emulation layer (qemu).
+
+    Histrionically it is set to ``false`` by default, but everyone is
+    encouraged to enable it.
```
Thanks @mbakke for your review. I've fixed the typo and mentioned the new default for Ganeti-3.1.

rbott commented:
Looks good, please proceed @saschalucas :-)

rbott commented:
...ok one more question :-) 

Should we change this (a few lines below) as well to `True`?
https://github.com/ganeti/ganeti/blob/6b6e28276447880f85a37c046682e0154f056b43/lib/hypervisor/hv_kvm/monitor.py#L522-L526

rbott commented:
Thanks for researching the "mq" part - I should have done that myself :-(
